src/NelsonSiegelSvensson.jl

```diff
@@ -0,0 +1,181 @@
+abstract type ParametricModel end
+
+"""
+    NelsonSiegel(β₀, β₁, β₂, τ₁)
+
+Parameters of Nelson and Siegel (1987) parametric model. 
+
+# Examples
+
+```julia-repl
+julia> β₀, β₁, β₂, τ₁ = 0.6, -1.2, -1.9, 3.0
+julia> nsm = NelsonSiegelSvensson.NelsonSiegel.(β₀, β₁, β₂, τ₁)
+```
+"""
+struct NelsonSiegel <: ParametricModel
+    β₀
+    β₁
+    β₂
+    τ₁
+    #r² no need
+
+    #function NelsonSiegel(β₀, β₁, β₂, τ₁, r²)
+    function NelsonSiegel(β₀, β₁, β₂, τ₁)
+        #(τ₁ <= 0 || r² < 0 || r² > 1) && throw(DomainError("Wrong parameter ranges"))
+        (τ₁ <= 0) && throw(DomainError("Wrong parameter ranges"))
+        #return new(β₀, β₁, β₂, τ₁, r²)
+        return new(β₀, β₁, β₂, τ₁)
+    end
+end
+
+"""
+    NelsonSiegelSvensson(β₀, β₁, β₂, β₃, τ₁, τ₂)
+
+Parameters of Svensson (1994) parametric model. 
+
+# Examples
+
+```julia-repl
+julia> β₀, β₁, β₂, β₃, τ₁, τ₂ = 0.6, -1.2, -2.1, 3.0, 1.5
+julia> nssm = NelsonSiegelSvensson.NelsonSiegelSvensson.(β₀, β₁, β₂, β₃, τ₁, τ₂)
+```
+"""
+struct NelsonSiegelSvensson <: ParametricModel
+    β₀
+    β₁
+    β₂
+    β₃
+    τ₁
+    τ₂
+    #r² no need
+
+    #function NelsonSiegelSvensson(β₀, β₁, β₂, β₃, τ₁, τ₂, r²)
+    function NelsonSiegelSvensson(β₀, β₁, β₂, β₃, τ₁, τ₂)
+        #(τ₁ <= 0 || τ₂ <= 0 || r² < 0 || r² > 1) && throw(DomainError("Wrong parameter ranges"))
+        (τ₁ <= 0 || τ₂ <= 0) && throw(DomainError("Wrong parameter ranges"))
+        #return new(β₀, β₁, β₂, β₃, τ₁, τ₂, r²)
+        return new(β₀, β₁, β₂, β₃, τ₁, τ₂)
+    end
+end
+
+Base.zero(ns::NelsonSiegel, t) = Continuous(ns.β₀ .+ ns.β₁ .* (1.0 .- exp.(-t ./ ns.τ₁)) ./ (t ./ ns.τ₁) .+ ns.β₂ .* ((1.0 .- exp.(-t ./ ns.τ₁)) ./ (t ./ ns.τ₁) .- exp.(-t ./ ns.τ₁)))
```

Comment:
The [docs](https://juliaactuary.github.io/Yields.jl/stable/api/#Base.zero-Union{Tuple{YC},%20Tuple{YC,%20Any}}%20where%20YC%3C:Yields.AbstractYield) for `zero` would suggest that `zero(ns::NelsonSiegel,t)` should return a `Periodic(1)` compounding frequency.

The options: 
- `convert(Periodic(1),ns_rate)` to make it consistent
  - and define a `zero(ns::NelsonSiegel,t,c::Continuous)` method for users who want the continuous rate
- We could update the docs - in some sense it doesn't really matter what's returned by default because the `Rate` type conveys that information

I'm open to either option - would returning a `Periodic(1)` here be surprising to users?

Reply:
@kasperrisager any thoughts on this? Abide by returning `Periodic(1)` or just let the rates speak for themselves?

Reply:
Thank you both. Please advise which way(s) to go, and sure I would love to get a contributor sticker at leeyuntien@gmail.com.